src/pages/en/guides/content-collections.mdx

```diff
@@ -396,6 +396,127 @@ const { Content } = await entry.render();
+
+2. [Set up TypeScript](/en/guides/content-collections/#setting-up-typescript) for content collections.
+3. Move Markdown and MDX pages from `src/pages` to subdirectories of `src/content`. Each subdirectory represents a content type. For example, to migrate the [blog posts in the tutorial](/en/tutorial/2-pages/2/), move the contents of `src/pages/posts` to `src/content/posts`. 
+4. Create a `src/content/config.ts` file and [define a schema](/en/guides/content-collections/#defining-a-collection-schema) for each content type. For the blog, we only have one content type, `posts`:
+
+```ts title="src/content/config.ts"
+// Import utilities from `astro:content`
+import { z, defineCollection } from "astro:content";
+// Define a schema for each collection you'd like to validate.
+const postsCollection = defineCollection({
+    schema: z.object({
+      title: z.string(),
+      pubDate: z.date(),
+      description: z.string(),
+      author: z.string(),
+      image: z.object({
+        url: z.string(),
+        alt: z.string()
+      }),
+      tags: z.array(z.string())
+    })
+ });
+// Export a single `collections` object to register your collection(s)
+export const collections = {
+  posts: postsCollection,
+};
+```
+
+:::tip
+If your editor doesn't recognize `astro:content`, make sure you're on the latest version of Astro and try restarting the dev server. 
+:::
+
+5. To [generate routes](/en/guides/content-collections/#generating-routes-from-content), create a `.astro` page with a [dynamic route](/en/core-concepts/routing/#dynamic-routes) in `src/pages/posts` to [query and render your content](#querying-collections). Our tutorial blog uses the static output mode, so we'll use `getStaticPaths()` to create a route for each post:
+
+```astro title="src/pages/posts/[...slug].astro"
+---
+import { getCollection } from 'astro:content';
+import MarkdownPostLayout from '../../layouts/MarkdownPostLayout.astro';
+
+export async function getStaticPaths() {
+  const blogEntries = await getCollection('posts');
+  return blogEntries.map(entry => ({
+    params: { slug: entry.slug }, props: { entry },
+  }));
+}
+
+const { entry } = Astro.props;
+const { Content } = await entry.render();
+---
+<MarkdownPostLayout frontmatter={entry.data}>
+  <Content />
+</MarkdownPostLayout>
+```
+
+6. Since we wrapped the content in a layout in the dynamic route, remove the `layout` definition in each post's frontmatter.
+
+```md title="src/content/post-1.md" del={2}
+---
+layout: ../../layouts/MarkdownPostLayout.astro
+title: 'My First Blog Post'
+pubDate: 2022-07-01
+...
+---
+```
+
+7. Replace `Astro.glob()` with [`getCollection()`](/en/reference/api-reference/#getcollection) to fetch content and metadata from your Markdown files. You will also need to update references to the returned post object, since you will now find your frontmatter values on the `data` property.
+
+For example, our unique tags logic becomes:
```

Comment:
Since this example is very close to the dynamic route page from before, maybe before this we could show the "blog index" page converted? Something where we just use `getCollection()` in order to create the linked list of posts? I don't think we need to remove this one, since it feels like an extra "Hey, think of all the places you might have used this!" But, I think maybe starting with an example of blog index makese sense, both for variety and for an example of something different.

Reply:
Added all the pages!